q2_feature_table/_filter.py

```diff
@@ -80,9 +81,19 @@ def filter_features(table: biom.Table, min_frequency: int=0,
                     metadata: qiime2.Metadata=None, where: str=None,
                     exclude_ids: bool=False)\
                    -> biom.Table:
-    _filter(table=table, min_frequency=min_frequency,
-            max_frequency=max_frequency, min_nonzero=min_samples,
-            max_nonzero=max_samples, metadata=metadata,
-            where=where, axis='observation', exclude_ids=exclude_ids)
+    _filter_table(table=table, min_frequency=min_frequency,
+                  max_frequency=max_frequency, min_nonzero=min_samples,
+                  max_nonzero=max_samples, metadata=metadata,
+                  where=where, axis='observation', exclude_ids=exclude_ids)
 
     return table
+
+
+def filter_seqs(data: pd.Series, metadata: qiime2.Metadata, where: str=None,
```

Comment:
As implemented, this feature performs id-based filtering (with or without a `where` clause), so it made sense to me to make `metadata` required.

Reply:
Looks good, this matches how `qiime diversity filter-distance-matrix` works (it only performs ID- and metadata-based filtering).

Reply:
Should this method be called `filter_seq_data` to match the merge operations on non-table objects in this plugin?

Reply:
@ebolyen, i'm fine with that, @jairideout, what do you think?

Reply:
I like `filter-seqs` because it matches `tabulate-seqs` and is shorter than `filter-seq-data`. The `-data` suffix doesn't really add much here, except for promoting arthritis!

I'd be in favor of dropping `-data` from `merge-seq-data` and `merge-taxa-data`, though that's outside the scope of this PR (may be a good change to make soon if others are on board with it, since it's user-facing). If we went with this strategy, the new method names could be `merge-seqs` and `merge-taxa`, respectively.

Reply:
I like that plan!